x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals/helpers.ts

```diff
@@ -71,7 +77,9 @@ export const replaceTemplateFieldFromQuery = (query: string, ecsData: Ecs) => {
     const valueToChange = findValueToChangeInQuery(esKuery.fromKueryExpression(query));
     return valueToChange.reduce((newQuery, vtc) => {
       const newValue = get(vtc.field, ecsData);
-      if (newValue != null) {
+      if (hasValueInStringArray(newValue)) {
+        return newQuery.replace(vtc.valueToChange, newValue[0]);
+      } else if (newValue != null) {
```

Comment:
IMO, we need to add a unit test for this bug fix to be accepted. Especially if the suggestion is for it to be back-ported to 7.6.2.